main.js

```diff
@@ -71,6 +90,19 @@ app.use(views(__dirname + '/templates', {
   }
 }));
 
+app.use(async (ctx, next) => {
+  try {
+    await next()
+  } catch (e) {
+    if (e instanceof NotFound) {
+      ctx.status = 404;
+      await ctx.render('404', {});
+    } else {
+      throw e;
+    }
+  }
+});
```

Comment:
Ooh, maybe do the same thing here: put this middleware in it's own file? This makes for self-documenting code too.